src/acat/azext_acat/custom.py

```diff
@@ -0,0 +1,538 @@
+    class CreateAcatReportWithDupAadToken(_AcatCreateReport.ReportCreateOrUpdate):
+        CLIENT_TYPE = "DupAadTokenClient"
+
+    def _execute_operations(self):
+        self.pre_operations()
+        yield self.CreateAcatReportWithDupAadToken(ctx=self.ctx)()
+        self.post_operations()
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        from azure.cli.core.aaz import AAZDateTimeArg, AAZStrArg
+
+        args_schema.time_zone._required = False
+        args_schema.time_zone._registered = False
+        args_schema.time_zone_with_default = AAZStrArg(
+            options=["--time-zone"],
+            arg_group="Properties",
+            help="""Report collection trigger time\'s time zone, the available list can be obtained by executing
+            "Get-TimeZone -ListAvailable" in PowerShell. An example of valid timezone id is "Pacific Standard Time".""",
+            default=GetLocalTimeZone(),
+        )
+        args_schema.trigger_time._required = False
+        args_schema.trigger_time._registered = False
+        args_schema.trigger_time_with_default = AAZDateTimeArg(
+            options=["--trigger-time"],
+            arg_group="Properties",
+            help="Report collection trigger time.",
+            default=GetClosestFullHour(),
+        )
+        return args_schema
+
+    def pre_operations(self):
+        cmd = OnboardAcat(cli_ctx=self.cli_ctx)
+
+        poller = cmd._handler(
+            {
+                "cmd": cmd,
+                "subscription_ids": ParseSubsFromResources(self.ctx.args.resources),
+            }
+        )
+        LongRunningOperation(cmd.cli_ctx, f"Starting {cmd.name}")(poller)
+
+        args = self.ctx.args
+        args.time_zone = args.time_zone_with_default
+        args.trigger_time = args.trigger_time_with_default
+
+
+class UpdateAcatReport(_AcatUpdateReport):
+    class UpdateAcatReportWithDupAadToken(_AcatUpdateReport.ReportUpdate):
+        CLIENT_TYPE = "DupAadTokenClient"
+
+    def _execute_operations(self):
+        self.pre_operations()
+        yield self.UpdateAcatReportWithDupAadToken(ctx=self.ctx)()
+        self.post_operations()
+
+
+class DeleteAcatReport(_AcatDeleteReport):
+    class DeleteAcatReportWithDupAadToken(_AcatDeleteReport.ReportDelete):
+        CLIENT_TYPE = "DupAadTokenClient"
+
+    def _execute_operations(self):
+        self.pre_operations()
+        yield self.DeleteAcatReportWithDupAadToken(ctx=self.ctx)()
+        self.post_operations()
+
+
+class DownloadAcatReport(_AcatDownloadSnapshot):
+    class DownloadAcatReportWithDupAadToken(_AcatDownloadSnapshot.SnapshotDownload):
+        CLIENT_TYPE = "DupAadTokenClient"
+
+    def _execute_operations(self):
+        self.pre_operations()
+        if self.ctx.args.snapshot_name is None:
+            yield f"No snapshot found for report {self.ctx.args.report_name}"

```

Comment:
Do you want to show it as warning or to raise it as error with the following logics stopped?

Reply:
Yes. It's better. But the validation of argument logic should be in pre_operations hook